File: 0528-swapping-nodes-in-a-linked-list/0528-swapping-nodes-in-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
 int getLength(struct ListNode* head) {
    int length = 0;
    while (head != NULL) {
        length++;
        head = head->next;
    }
    return length;
}
struct ListNode* swapNodes(struct ListNode* head, int k) {
    int n = getLength(head);
    int frontindex = k-1;
    int backindex = n-k;

    struct ListNode* frontnode = head;
    struct ListNode* backnode = head;

    for(int i=0;i<frontindex ;i++){
        frontnode = frontnode->next;
    }

    for(int i=0;i<backindex ;i++){
        backnode = backnode->next;
    }

    int temp = frontnode->val;
    frontnode->val = backnode->val;
    backnode->val = temp; 

    return head;

}

File: 0528-swapping-nodes-in-a-linked-list/0528-swapping-nodes-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
         int n = getLength(head);
        int frontIndex = k - 1;
        int backIndex = n - k;
        
        ListNode frontNode = head;
        ListNode backNode = head;
        
        for (int i = 0; i < frontIndex; i++) {
            frontNode = frontNode.next;
        }
        
        for (int i = 0; i < backIndex; i++) {
            backNode = backNode.next;
        }
        
        int temp = frontNode.val;
        frontNode.val = backNode.val;
        backNode.val = temp;
        
        return head;
    }
    public static int getLength(ListNode head) {
        int length = 0;
        ListNode current = head;
        while (current != null) {
            length++;
            current = current.next;
        }
        return length;
    }
}